CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+    println(msg);
+
+}
+```
+
+Which can just be simplified to
+
+```
+void Commands::setTto(bool throttleTimeOut) {
+    String msg = "ASTRO Throttle Timeout " + String(throttleTimeOut ? "On" : "Off");
+    println(msg);
+
+}
+```
+
+What we gain by eliminating duplication is the need to change the function multiple times to make changes. If the logic above need to add another instruction, we would need to change `ttoOn(void)` and `ttoOff(void)` while now we only need to change `setTto(bool)`
+
+There are more changes we could do to improve this function that we will see later.
+
+
+### 2. Coupling
+Coupling is the degree of interdependence between software modules.
+
+#### 2.1 Modularity
+Since the rover needs to be modular, we need to have very loose coupling. The software needs to have independent modules that we can quickly swap out. This is one of the many reasons we use ROS, which creates a modular foundation.
+
+#### 2.2 Levels of abstraction
+It is a best practice to abstract the code into different layers. For example, code related to communication with ROS should not include GUI changes. Common layers are GUI, Logic, Communication.
+
+
+### 3. Comments
+Comments can be a blessing or a cure. Proper documentation can help developers while bad documentation can obstruct or mislead them.
+
+#### 3.1 Useless comments
+Ideal code is self explanatory. Do not comment when it is obvious like
+
+```
+//Initialize ROS
+initRos();
+```
+
+This adds lines for the programmers to read without adding any value.
+
+#### 3.2 Rotted Comments
+One major issue with comments is that they rot with time. Before writing a comment, make sure that this comment will still be relevant after the code passes through a lot of changes.
+
+#### 3.3 Todos
+Do not write any `Todo` comments. These get easily lost and forgotten. Open an issue instead.
+
+#### 3.4 Commented code
+Avoid commenting code. A lot of the code that is commented, never gets uncommented again. It is a better practice to remove the code. We are using git as a version control tool, this means that old code can always be recuperated.
+ 
+
+### 4. Meaningful Names
+
+#### 4.1 Variable Names
+
+With the function that we improved above there is still a glaring issue :
+
+```
+void Commands::setTto(bool throttleTimeOut) {
+    String msg = "ASTRO Throttle Timeout " + String(throttleTimeOut ? "On" : "Off");
+    println(msg);
+
+}
+```
+
+What the heck is `TTO`? By observing this function and it's argument, it is obvious that it's for `Throttle Timeout`. However, imagine that you were stumbling upon a call to this function like this
+
+```
+//some code
+setTto(false);
+//more code
+```
+To know what `TTO` means, you need to go to the function definition which could be tedious if you have to do this often.
+
+Furthermore, you would expect `throttleTimeOut` to be a number, the timeout time, but it is a boolean.
+
+```
+void Commands::hasThrottleTimeout(bool hasThrottleTimeout) {
```

Comment:
should the function be named with `has` at the front? It does not return a boolean, just basically prints text based off the value of a given boolean.

Reply:
true, it sets something though